frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.unit.spec.tsx

```diff
@@ -0,0 +1,137 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import type { Engine } from "metabase-types/api";
+
+import DatabaseForm from "./DatabaseForm";
+
+interface SetupOpts {
+  onSubmit?: () => void;
+}
+
+function setup({ onSubmit }: SetupOpts) {
+  render(<DatabaseForm engines={engines} isAdvanced onSubmit={onSubmit} />);
+}
+
+describe("DatabaseForm", () => {
+  it("should submit default values", async () => {
+    const mockOnSubmit = jest.fn();
+    setup({ onSubmit: mockOnSubmit });
+
+    const expectedDatabaseName = "My H2 Database";
+    const expectedConnectionString = "file:/somewhere";
+    userEvent.type(screen.getByLabelText("Display name"), expectedDatabaseName);
+    userEvent.type(
+      screen.getByLabelText("Connection String"),
+      expectedConnectionString,
+    );
+
+    const saveButton = screen.getByRole("button", { name: "Save" });
+    await waitFor(() => {
+      expect(saveButton).toBeEnabled();
+    });
+    userEvent.click(saveButton);
+
+    const expectedDefaultSchema = {
+      schedules: {
+        metadata_sync: undefined,
+        cache_field_values: undefined,
+      },
+      auto_run_queries: true,
+      refingerprint: false,
+      cache_ttl: null,
+      is_sample: false,
+      is_full_sync: true,
+      is_on_demand: false,
+    };
+
+    await waitFor(() => {
+      expect(mockOnSubmit).toHaveBeenCalledWith({
+        ...expectedDefaultSchema,
+        engine: "h2",
+        name: expectedDatabaseName,
+        details: {
+          "advanced-options": false,
+          db: expectedConnectionString,
+        },
+      });
+    });
+  });
+});
+
+const engines: Record<string, Engine> = {
+  h2: {
```

Comment:
the `engine` variable is copied from the storybook https://github.com/metabase/metabase/blob/master/frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.stories.tsx#L14